tests/test_shards_subset.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2022 Anaconda, Inc
+# Copyright (C) 2023 conda
+# SPDX-License-Identifier: BSD-3-Clause
+import pytest
+import pytest_codspeed
+from conda.common.compat import on_win
+from conda.models.channel import Channel
+
+from conda_libmamba_solver.shards import fetch_channels
+from conda_libmamba_solver.shards_subset import RepodataSubset, build_repodata_subset
+
+TESTING_SCENARIOS = [
+    {
+        "name": "python",
+        "packages": ["python"],
+        "prefetch_packages": [],
+        "channel": "conda-forge-sharded",
+        "platform": "linux-64",
+    },
+    {
+        "name": "data-science-basic",
+        "packages": ["numpy", "pandas", "matplotlib"],
+        "prefetch_packages": ["python"],
+        "channel": "conda-forge-sharded",
+        "platform": "linux-64",
+    },
+    {
+        "name": "data-science-ml",
+        "packages": ["scikit-learn", "matplotlib"],
+        "prefetch_packages": ["python", "numpy"],
+        "channel": "conda-forge-sharded",
+        "platform": "linux-64",
+    },
+    {
+        "name": "web-development",
+        "packages": ["django", "celery"],
+        "prefetch_packages": ["python", "requests"],
+        "channel": "conda-forge-sharded",
+        "platform": "linux-64",
+    },
+    {
+        "name": "scientific-computing",
+        "packages": ["scipy", "sympy", "pytorch"],
+        "prefetch_packages": ["python", "numpy", "pandas"],
+        "channel": "conda-forge-sharded",
+        "platform": "linux-64",
+    },
+    {
+        "name": "devops-automation",
+        "packages": ["ansible", "pyyaml", "jinja2"],
+        "prefetch_packages": ["python"],
+        "channel": "conda-forge-sharded",
+        "platform": "linux-64",
+    },
+]
+
+
+def codspeed_supported():
+    """
+    TODO: temporary measure to skip these tests if we do not have pytest-codspeed >=4
+    """
+    try:
+        major, minor, bug = pytest_codspeed.__version__.split(".")
+        return int(major) >= 4
+    except (ValueError, AttributeError):
+        # If this fails, it means we want to skip this test
+        return False
+
+
+def clean_cache(conda_cli):
+    """
+    Clean cache and assert it completed without error except on Windows
+    """
+    out, err, _ = conda_cli("clean", "--yes", "--all")
+
+    # Windows CI runners cannot reliably remove this file, so we don't care about
+    # this assertion on that platform
+    if not on_win:
+        assert not err
```

Comment:
err contains debugging output in the vscode test runner. Pull or cherry-pick https://github.com/dholth/conda-libmamba-solver/commit/9d86cec8852cf75319c9b6da0c9f88f26712946a for a fix, or I can try to push it here directly.